src/Credentials/CredentialProvider.php

```diff
@@ -692,6 +706,35 @@ private static function loadDefaultProfiles() {
         return $profiles;
     }
 
+    public static function getAssumeRoleCredentials(
+        $sourceProfileName = '',
+        $profileName = '',
+        $filename = '',
+        $config = []
+    )
+    {
+        $data = self::loadProfiles($filename);
+        $credentialSource = !empty($data[$profileName]['credential_source']) ? $data[$profileName]['credential_source'] : null;
+        if (isset($credentialSource) && is_string($credentialSource)) {
```

Comment:
Do you need the `is_string`, considering you have a default exception throw in the switch below?